src/Beacon.jsx

```diff
@@ -157,6 +164,23 @@ export class Beacon extends Component {
         this.setState({ targetElement, targetBounds }); // eslint-disable-line react/no-did-update-set-state
       }
     }
+
+    if (this.state.tooltip && this.state.targetBounds && !this.state.arrowPosition) {
+      // Calculate relative arrow position when ref is available
+      setTimeout(() => {
+        const arrowStyleLeft = this.getArrowHorizontalPosition(
+          this.state.targetBounds,
+          this.refs['tooltip'].getBoundingClientRect(),
+          this.refs['tooltipArrow'].getBoundingClientRect()
+        );
+        this.setState({
+          arrowPosition: (this.state.tooltipAttachmentVertical === 'top' ||
+            this.state.tooltipAttachmentVertical === 'bottom') ?
+            { left: arrowStyleLeft, right: 'auto' } :
```

Comment:
We only calculate the position when the arrow slides horizontally, it is otherwise always centered for `middle` position.